CameraXVideo/app/src/main/java/com/example/android/camerax/video/fragments/CameraFragment.kt

```diff
@@ -97,15 +100,24 @@ class CameraFragment : Fragment() {
         val cameraProvider = ProcessCameraProvider.getInstance(requireContext()).await()
 
         val cameraSelector = getCameraSelector(cameraIndex)
-        val preview = Preview.Builder().setTargetAspectRatio(DEFAULT_ASPECT_RATIO)
-            .build().apply {
-                setSurfaceProvider(fragmentCameraBinding.previewView.surfaceProvider)
-            }
 
         // create the user required QualitySelector (video resolution): we know this is
         // supported, a valid qualitySelector will be created.
-        val qualitySelector = QualitySelector.of(
-            cameraCapabilities[cameraIndex].qualitySelector[qualitySelectorIndex])
+        val quality = cameraCapabilities[cameraIndex].qualitySelector[qualitySelectorIndex]
+        val qualitySelector = QualitySelector.of(quality)
+
+        fragmentCameraBinding.previewView.updateLayoutParams<ConstraintLayout.LayoutParams> {
+            val orientation = this@CameraFragment.resources.configuration.orientation
+            dimensionRatio = qualitySelector.getAspectRatioString(quality,
+                (orientation == Configuration.ORIENTATION_PORTRAIT))
+        }
+
+        val preview = Preview.Builder()
+            .setTargetAspectRatio(qualitySelector.getAspectRatio(quality))
+            .setTargetRotation(fragmentCameraBinding.previewView.display.rotation)
```

Comment:
We don't need to set target rotation for preview. The default rotation should be good enough. You only need to set the target rotation if you want to do something fancy. e.g. like upside-down preview.

On the other hand, the target rotation of VideoCapture is more interesting. The default rotation won't work, if the app/device's orientation is locked. e.g. locked portrait mode. You need to set it using [sensor orientation](https://developer.android.com/reference/androidx/camera/view/RotationProvider ).   You need to do the same thing for ImageCapture and/or imageAnalysis.

If you are wondering why preview is special, it is because preview is displayed in the same orientation so they cancel each other out.  CameraController provide features to handle this.

Reply:
great point: removed it from preview build; will add it into the video capture use case builder when the rotation update API is released as public API in the future.